### PR TITLE
chore: Make temporary solution for API key [DEV-3400]

### PR DESCRIPTION
**Pull request by anikitinDSR**
Added:
- small refactoring for API guarding
- add endpoint for getting idToken
- add apiKey table and tracking API keys

### CODE REVIEW

**Comment by ankurdotb** [comment]
Task linked: [DEV-3400 Add `makeFeePayment` parameter for `/credential/verify` and for `/presentation/verify`](https://app.clickup.com/t/6600954/DEV-3400)

File: src/middleware/auth/types.ts
Comment (Eengineer1) [inline src/middleware/auth/types.ts:1]: Looks empty, so let's remove.

File: src/middleware/auth/user-info-fetcher/api-token.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+			}
+			const scopes = await oauthProvider.getScopesForRoles(payload.roles as string[]);
+			if (!scopes) {
+				return this.returnError(StatusCodes.UNAUTHORIZED, `Unauthorized error: No scopes found for the roles.`);
+			}
+			this.setScopes(scopes);
+			this.setUserId(payload.sub as string);
+			return this.returnOk();
+		} catch (error) {
+			return this.returnError(StatusCodes.UNAUTHORIZED, `Unauthorized error: ${error}`);
```
Comment (Eengineer1) [inline src/middleware/auth/user-info-fetcher/api-token.ts:48]: Is this catch-all block intentionally marking all possible caught errors as client originating?

Here all exceptions will end up responding with 401 unauthorised.

**Review by Eengineer1 — CHANGES_REQUESTED** [review]
Looks good.

Left remarks considering error handling edge cases to resolve, other than that, that's good to go.

**Pull request merged**